.github/workflows/docker.yml

```diff
@@ -0,0 +1,64 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ '*' ]
+
+env:
+  IMAGE_NAME: rusty
+
+jobs:
+  # Push image to GitHub Packages.
+  # See also https://docs.docker.com/docker-hub/builds/
+  build-linux:
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      matrix:
+        config:
+          - {
+            os: "ubuntu-latest",
+            version: "linux",
+            arch: "x86_64"
+          }
+    permissions:
+      packages: write
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Build image
+        shell: bash
+        run: docker buildx build . --platform ${{matrix.config.version}}/${{matrix.config.arch}}  --file Dockerfile --tag $IMAGE_NAME
+
+      - name: Log in to registry
+        if: ${{ github.event_name != 'pull_request' }}
+        # This is where you will update the PAT to GITHUB_TOKEN
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Push image
+        shell: bash
+        if: ${{ github.event_name != 'pull_request' }}
+        run: |
+          IMAGE_ID=ghcr.io/${{ github.repository_owner }}/$IMAGE_NAME
+
+          # Extract branch name
+          BRANCH_NAME=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+
+          # Append branch name to image ID
+          IMAGE_ID=$IMAGE_ID-$BRANCH_NAME
+
+          # Change all uppercase to lowercase
+          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
+
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "main" ] && VERSION=latest
+          #Add the platform to the version
+          VERSION=$VERSION-${{ matrix.config.arch }}
+          echo IMAGE_ID=$IMAGE_ID
+          echo VERSION=$VERSION
+          docker tag $IMAGE_NAME $IMAGE_ID:$VERSION
+          docker push $IMAGE_ID:$VERSION
```

Comment:
This ends up only pushing images like : latest-amd64 instead of latest.
To add the  latest tag, you can use a manifest (see: https://github.com/PLC-lang/rust-llvm-images/blob/db945c7087d6f0ef51b88222702fe19538a7fbd1/.github/workflows/docker.yml#L99)
This would support the multiplatform docker images where it will pull the correct image based on platform.